### DISCUSSION TITLE
Custom text input box size

### DISCUSSION THREAD

**Discussion by compemperor** [open-webui/open-webui#7430] [Ideas]
# Feature Request

Would like to be able to customize the size of the chat input box. For example: 
![image](https://github.com/user-attachments/assets/cee0d72f-215c-4fe6-9bc8-16495bfaed79)

Right now I solved the size issue via Nginx sub_filter, it would have been nice to be able to change the input box size in the Admin settings. 

` #filters

sub_filter_types *;
sub_filter '</head>' '<style type="text/css" nonce>#chat-input {height: 150px !important; }  </style> </head>';
sub_filter_once on;`

.

**Comment by Chlorek** [comment]
I am new to the open-webui, so not sure what the best workflow here can be, but I was just looking for such a feature. When working with bigger prompts containing for example parts of documents or source code files it becomes tedious to navigate it and change something if necessary later. I can also picture it as really useful when editing existing messages.
Of course one can start preparing prompt in separate text editor first, but to make workflow better making the text field expandable sounds like a good idea to me, probably to even bigger sizes than in the suggestion screenshot.

**Comment by aindriu80** [comment]
The issue is that the text box size doesn't increase as you type - i.e. more than one line of text visible at any one time.  Often I need to read back the last few lines of text before hitting Enter, a dynamically increasing text box (within reason) would be great.

You can achieve a dynamically increasing text box using pure CSS with `textarea` and `height: auto`, combined with JavaScript to adjust the height as the user types:-

### Solution:
Use CSS to allow expansion and JavaScript to dynamically resize the `textarea`.

#### **CSS**
```css
textarea {
  width: 100%; /* Adjust as needed */
  min-height: 40px; /* Start with a reasonable height */
  max-height: 300px; /* Prevent it from growing too large */
  overflow-y: hidden; /* Hide scrollbars */
  resize: none; /* Prevent manual resizing */
}
```

#### **JavaScript**
```javascript
document.addEventListener("DOMContentLoaded", function () {
  const textarea = document.querySelector("textarea");

  textarea.addEventListener("input", function () {
    this.style.height = "auto"; // Reset height
    this.style.height = this.scrollHeight + "px"; // Set new height
  });
});
```

### **How It Works:**
- The CSS ensures the `textarea` starts at a small height, disables manual resizing, and hides the scrollbar.
- The JavaScript listens for user input and dynamically sets the `height` based on `scrollHeight`, which grows as more text is added.

**Reply by Classic298** [reply]
Cool, can you do a PR?

**Reply by gurrunyuflo** [reply]
That looks like exactly what its needed! But where is this code to be inserted?
Why could this not be the default behavior? I think it is much more convenient than having a single line of text shown in the input box after the first model answer.
